src/docs/asciidoc/modules/ROOT/pages/chapter4.adoc

```diff
@@ -142,394 +214,537 @@ entity Tag {
-
-----
-docker-compose -f gateway/src/main/docker/jhipster-registry.yml restart
-----
+=== Zero turnaround development that sparks joy
 
-To log in, you'll need to add `http://localhost:8761/login` as a login-redirect URI in your Okta app.
+At this point, I've only shown you how to run the Spring Boot backends with their packaged React apps. What if you want to work on the UI and have zero turnaround that sparks joy? ✨🤗
 
-== Run your microservices architecture
+In the gateway app, run `npm start`. This command will run the UI on a web server, open a browser window to `\http://localhost:9000`, and use https://browsersync.io/[Browsersync] to keep your browser in sync with your code.
 
-Open three terminal windows and navigate to each app (`gateway`, `blog`, and `store`). In each window, and run Maven to start each app:
+Modify the code in `gateway/src/main/webapp/app/modules/home/home.tsx` to make a quick change. For example, add the following HTML below the `<h2>`.
 
-[source,java]
+[source,html]
 ----
-./mvnw
+<h3 className="text-primary">
+  Hi, I'm a quick edit!
+</h3>
 ----
 
-TIP: If you already have Maven installed, you can simply use `mvn`.
-
-Open your browser and go to http://localhost:8761. Log in, and you should see a welcome page that shows that the gateway and both apps have been registered.
+You'll see this change immediately appear within your browser.
 
-.JHipster Registry with gateway registered
-image::chapter4/jhipster-registry-with-gateway+apps.png[JHipster Registry with gateway registered, 800, scaledwidth=100%, align=center]
+.Gateway quick edit
+image::chapter4/gateway-quick-edit.png[Gateway quick edit, 1132, scaledwidth=100%, align=center]
 
-Once everything is running, open a browser, go to http://localhost:8080, and click *sign in*. You should be redirected to your Okta tenant to sign in, then back to the gateway once you've entered valid credentials.
+Remove it, and it'll disappear right away too.
 
-.Welcome, JHipster
-image::chapter4/welcome-jhipster.png["Welcome, JHipster", 1200, scaledwidth=100%, align=center]
+Now, open another terminal and navigate into the `store` directory. Run `npm start`, and you'll have a similar zero-turnaround experience when modifying files in the `store` app. The app will start a webserver on `\http://localhost:9002`, and there will only be one menu item for product. Modify files in the `store/src/main/webapp/app/entities/store/product` directory, and you'll see the changes in your browser immediately. For example, change the wrapper `<div>` in `product.tsx` to have a background color:
 
-.Okta sign-in
-image::chapter4/okta-sign-in.png[Okta sign-in, 1200, scaledwidth=100%, align=center]
+[source,html]
+----
+<div className="bg-info">
+----
 
-.JHipster after Okta SSO
-image::chapter4/jhipster-logged-in.png[JHipster after Okta SSO, 800, scaledwidth=100%, align=center]
+The UI will change before you can kbd:[Cmd+Tab] back to your browser.
```

Comment:
Is `kbd` a well known initalism? perhaps a link to the [mozilla](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/kbd) docs page would be good? I just heard about kbd as I'm mostly a back end developer and haven't encountered kbd before